app/modules/entities/lib/entities.js

```diff
@@ -115,6 +116,31 @@ export async function getEntitiesAttributesByUris ({ uris, attributes, lang, rel
   return { entities }
 }
 
+export async function getAndAssignPopularity (entities) {
+  const uris = []
+  let wdUrisCount = 0
+  entities.forEach(entity => {
+    const { uri } = entity
+    if (entity.popularity === undefined) {
+      uris.push(uri)
+      if (uri.startsWith('wd:')) wdUrisCount++
+    }
+  })
+  if (!isNonEmptyArray(uris)) return entities
+  // Limiting refresh to not overcrowd Wikidata
+  const refresh = wdUrisCount < 30
+  const urisChunks = chunk(uris, 30)
+  const responses = await Promise.all(urisChunks.map(async urisChunk => {
+    return preq.get(app.API.entities.popularity(urisChunk, refresh))
+  }))
+  const scores = mergeResponsesObjects(responses, 'scores')
+
+  entities.forEach(entity => {
+    const popularity = scores[entity.uri]
+    if (popularity >= 0) entity.popularity = popularity
+  })
+}
+
```

Comment:
Since those popularity are heavy for the server to lift, it make sense to let the client in control of the amount of popularity to request. This chunk of code above is not pretty but it does its job at only requesting unpopulated "popularities" without asking too much on wikidata entities.

Reply:
Unless the request to the server is made with `refresh=true`, the server will only return popularities already in cache ("dry" cache request)

Reply:
But if no popularity is cached, then server is doing the heavy lifting of requesting an entity graph, right ?

Reply:
see https://github.com/inventaire/inventaire/pull/686#issuecomment-1814552673

Reply:
Switching to `app.API.entities.getAttributesByUris` would drop the refresh possibility from the client, leaving responsibility to the server, right ?
Is it okay that client's `getAttributesByUris` dont not implement any refresh possibility then ?
It's quite some time and work on already written and functional code, what about leaving as it is for the moment, merge server PR and continue future work with this new attribute possibility in mind ?